client/X11/xf_cliprdr.c

```diff
@@ -1343,8 +1361,25 @@ static UINT xf_cliprdr_server_format_data_response(CliprdrClientContext*
 		}
 	}
 
+	/* Cache converted and original data to avoid doing a possibly costly
+	 * conversion again on subsequent requests */
 	clipboard->data = pDstData;
 	clipboard->data_length = DstSize;
+
+	/* We have to copy the original data again, as pSrcData is now owned
+	 * by clipboard->system. Memory allocation failure is not fatal here
+	 * as this is only a cached value. */
+	clipboard->data_raw = (BYTE*) malloc(size);
+	if (clipboard->data_raw)
+	{
+		CopyMemory(clipboard->data_raw, data, size);
+		clipboard->data_raw_length = size;
+	}
+	else
+	{
+		WLog_WARN(TAG, "failed to allocate %zu bytes for a copy of raw clipboard data", size);
```

Comment:
Please see https://github.com/FreeRDP/FreeRDP/blob/master/docs/PrintFormatSpecifiers.md regarding the print format specifiers.